apps/web/modules/settings/billing/components/BillingCredits.tsx

```diff
@@ -0,0 +1,126 @@
+import { ProgressBar } from "@tremor/react";
+import { useRouter } from "next/navigation";
+import { useForm } from "react-hook-form";
+
+import { IS_SMS_CREDITS_ENABLED } from "@calcom/lib/constants";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { useParamsWithFallback } from "@calcom/lib/hooks/useParamsWithFallback";
+import { trpc } from "@calcom/trpc/react";
+import { Button } from "@calcom/ui/components/button";
+import { TextField, Label, InputError } from "@calcom/ui/components/form";
+import { showToast } from "@calcom/ui/toast";
+
+import { BillingCreditsSkeleton } from "./BillingCreditsSkeleton";
+
+export default function BillingCredits() {
+  const { t } = useLocale();
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm<{ quantity: number }>({ defaultValues: { quantity: 50 } });
+
+  const params = useParamsWithFallback();
+
+  const teamId = params.id ? Number(params.id) : undefined;
```

Comment:
I wonder if we use nuqs library that we have installed to handle this a bit nicer. We don't really use pages dir so not needed to have this fallback
```
import { useQueryState, parseAsInteger } from 'nuqs';
```

```suggestion
const [teamId] = useQueryState(
  'id', 
  parseAsInteger.withDefault(undefined)
);
```

Reply:
This doesn't work for me, does this read from the URL path?

Reply:
```
const [teamId, setTeamId] = useQueryState(
  'id', 
  parseAsInteger.withDefault(undefined)
);
```

it's supposed to read it from the URL, and set it back to the URL (when calling the setter).

How does it not work for you?